### ISSUE TITLE
User Profile (/me):  The upload component doesn't fit accessibility requirements

### ISSUE THREAD

**Issue by gabrielcaires** [Automattic/wp-calypso#101523]
### Context and steps to reproduce

1. The Upload button is not focusable.
2. The modal doesn't follow the a11y requirements https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/  (E.g lack of aria-label)
3. The photo editing buttons are not accessible by keyboard.

Suggestion:
Explore replacing the current component with some open source like https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/ 

### Site owner impact
All users

### Severity

Medium

### What other impact(s) does this issue have?
N/A

### If a workaround is available, please outline it here.
N/A

### Platform
Desktop and Mobile

**Comment by rcrdortiz** [comment]
These changes are now live.
